scripts/generate_docs.sc

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env amm
+
+import java.time.Instant
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+
+/**
+  * Docs generation process:
+  *
+  * 1. copy docs/docs and docs/_layouts/docs.html part from 1.3, 1.4 and 1.5 branches to the temp dirs
+  * 2. copy the latest release (1.5_ branch doc to the temp dir
+  * 3. generate the latest release docs
+  * 4. replace docs/docs with docs/docs from 1.3, 1.4, 1.5 and generate docs for the _sites/docs_version
+  */
+
+@main
+def main(tracked_by_git: Boolean = true) = {
+  if(tracked_by_git) {
+    //If the script is being tracked by git, we should create and run an untracked copy of it in order
+    // to avoid errors during branch checkout
+    cp.over(pwd/"generate_docs.sc", pwd/"generate_docs.temp")
+    %('amm, "generate_docs.temp", "--tracked_by_git", "false")
```

Comment:
What is the reason for this script? I don't quite understand it, sorry.

Reply:
so, we want to run this script when it's not being tracked by git because it breaks if we switch branches. Do do this, we create a temporary untracked copy and run it.